extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -424,10 +424,18 @@ class QueueProcessor {
     // Object metadata may contain multiple elements for a single part if
     // the part was originally copied from another MPU. Here we reduce the
     // locations array to a single element for each part.
-    _getLocations(sourceEntry) {
+    _getLocations(sourceEntry, log) {
         const locations = sourceEntry.getLocation();
         const reducedLocations = [];
         let partTotal = 0;
+        if (locations.some(
```

Comment:
It would add complexity because we will need to check both the current and next part then, because we're calling `getPartNumber()` for each of them, which requires the `dataStoreETag` to be set. I think it's simpler to just do a loop first for this reason, I don't think it will cost any significant additional CPU, because when there are many parts, it's a big object so it will take a lot of time anyways.